src/mon/MgrMonitor.cc

```diff
@@ -1318,7 +1331,50 @@ bool MgrMonitor::prepare_command(MonOpRequestRef op)
     if (!pending_map.modules.count(module)) {
       ss << "module '" << module << "' is not enabled";
     }
+    dout(8) << __func__ << " disabling module " << module << " from new " << dendl;
     pending_map.modules.erase(module);
+  } else if (prefix == "mgr module force disable") {
+    string mod;
+    cmd_getval(cmdmap, "module", mod);
+
+    bool confirmation_flag = false;
+    cmd_getval(cmdmap, "yes_i_really_mean_it", confirmation_flag);
+
+    if (mod.empty()) {
+      ss << "Module name wasn't passed!";
+      r = -EINVAL;
+      goto out;
+    }
+
+    if (pending_map.get_always_on_modules().count(mod) == 0) {
+      ss << "Always-on module named \"" << mod << "\" does not exist";
+      r = -EINVAL;
+      goto out;
+    }
+
+    if (pending_map.force_disabled_modules.count(mod)) {
+      ss << "Module \"" << mod << "\"is already disabled";
+      r = 0;
+      goto out;
+    }
+
+    if (!pending_map.modules.count(mod)) {
+      ss << "module '" << mod << "' is not enabled";
+      r = 0;
+      goto out;
+    }
+
+    if (!confirmation_flag) {
+      ss << "This is a potentially disruptive operation. If you are sure "
```

Comment:
Maybe something like "This operation will remove commands and disable operations that other Ceph utilities expect to be available. Do not continue unless your cluster is already experiencing an event and you have been advised to disable this module as part of troubleshooting. If you are sure that you wish to continue, run again with --yes-i-really-mean-it".
(I really want to include the part about losing functionality, as it's more explicit than "potentially disruptive".

Reply:
Done.